modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h

```diff
@@ -357,10 +357,16 @@ class IVW_MODULE_BRUSHINGANDLINKING_API BrushingAndLinkingManager : public Seria
     void setInvalidationLevels(std::vector<BrushingTargetsInvalidationLevel> invalidationLevels);
 
     /**
-     * propagates the modified state to all child managers and resets the state. Should only be
-     * called by the brushing and linking ports _after_ the process() function has been called.
+     * propagates the modified state to all child managers. Needs to be called by the brushing and
+     * linking ports _before_ the invalidation level is queried.
+     * @see getInvalidationLevel
      */
-    void markAsValid();
+    void propagateModifications();
```

Comment:
Would these ever be called by anything but the ports? 
Maybe we could make these private friend functions to the ports.

Reply:
If you, for whatever reason, have a manager between the ports, these functions need to be called as well in the correct order. Otherwise changes are not propagated downward.